### PR TITLE
Add an IdpGetAll endpoint, return meta-url and login-url

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

File: services/spar/test/Arbitrary.hs [hunk 1]
```diff
@@ -27,6 +27,11 @@ instance Arbitrary NewIdP where
     _nidpPublicKey  <- arbitrary
     pure $ NewIdP {..}
 
+instance Arbitrary IdPList where
+  arbitrary = do
+    _idplProviders <- arbitrary
+    pure $ IdPList {..}
```
Comment (fisx) [inline services/spar/test/Arbitrary.hs]:
```haskell
arbitrary = IdPList <$> arbitrary
```

but it's both good.

File: services/spar/spar.integration.yaml [hunk 2]
```diff
@@ -14,6 +14,9 @@ saml:
       surname: Girlfriend
       email: email:president@evil.corp
 
+metaUrl:        https://app.wire.com/sso/metadata         # corresponds to 'APIMeta'
```
Comment (fisx) [inline services/spar/spar.integration.yaml]: Should be under `spInfo`.  There are at least two metadata urls, one for IdP, one for SP, and it's not clear from the name here which is meant.

Also, should the comment be in the source code?  That's what it refers to.

**Comment by fisx** [comment]
I will continue review later, and probably make the changes to I can merge myself if you have no objections.

Looking really good so far, thanks!

**Comment by neongreen** [comment]
I 140% don't mind any changes

File: services/spar/src/Spar/Types.hs [hunk 3]
```diff
@@ -23,7 +23,40 @@ import URI.ByteString
...
+  , _idpeSPInfo :: SPInfo
+  }
+  deriving (Eq, Show, Generic)
+
+makeLenses ''IdPExtra
+deriveJSON deriveJSONOptions ''IdPExtra
+
+-- | A list of 'IdP's, returned by some endpoints. Wrapped into an object to
+-- allow extensibility later on.
+data IdPList = IdPList
```
Comment (tiago-loureiro) [inline services/spar/src/Spar/Types.hs]: I think that similarly to https://github.com/wireapp/wire-server/pull/402, we should make this an object

**Comment by neongreen** [comment]
Isn't it already one? If not, it should be, yes

**Comment by fisx** [comment]
All good (with my nit-picky changes).  I'm going to merge this into #395 myself now.

**Comment by fisx** [comment]
thanks!

**Comment by fisx** [comment]
(missing: idp-put endpoint; tests.  i'll add that now to the parent PR.)

**Pull request merged**